packages/venia-ui/lib/components/Header/storeSwitcher.js

```diff
@@ -0,0 +1,124 @@
+    const { mobileView } = props;
+    const talonProps = useStoreSwitcher({
+        getStoreConfig: GET_CONFIG_DATA
+    });
+
+    const {
+        handleSwitchStore,
+        availableStores,
+        isLoading,
+        storeMenuRef,
+        storeMenuTriggerRef,
+        storeMenuIsOpen,
+        handleTriggerClick
+    } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const className = mobileView ? classes.root_mobile : classes.root;
+    const menuClassName = storeMenuIsOpen ? classes.menu_open : classes.menu;
+    const triggerClassName = classes.storeSwitcherContainer;
+
+    let children = null;
+
+    if (isLoading) {
+        children = (
+            <Icon
+                size={24}
+                classes={{
+                    root: classes.loadingContainer,
+                    icon: classes.loading
+                }}
+                src={Loader}
+            />
+        );
+    }
+
+    if (availableStores) {
+        const hasMultipleStores =
+            Object.keys(availableStores).length > 1 || null;
+        let currentStoreName = null;
+        const stores = Object.keys(availableStores).map(storeCode => {
+            const isCurrentStore = availableStores[storeCode].is_current;
+            const storeName = availableStores[storeCode].storeName;
+            const activeIcon = isCurrentStore ? (
+                <Icon size={20} src={Check} />
+            ) : null;
+
+            if (isCurrentStore) {
+                currentStoreName = storeName;
+            }
+
+            return (
+                <li key={storeCode} className={classes.menuItem}>
+                    <button
+                        className={classes.menuItemButton}
+                        onClick={() => {
+                            handleSwitchStore(storeCode);
+                        }}
+                    >
+                        <span className={classes.content}>
+                            <span className={classes.text}>{storeName}</span>
+                            {activeIcon}
+                        </span>
+                    </button>
+                </li>
+            );
+        });
+
+        children = hasMultipleStores ? (
+            <div className={triggerClassName} ref={storeMenuTriggerRef}>
+                <button
+                    className={classes.trigger}
+                    aria-label={currentStoreName}
+                    onClick={handleTriggerClick}
+                >
+                    <Icon src={MapPin} />
+                    <span className={classes.label}>{currentStoreName}</span>
+                </button>
+                <div ref={storeMenuRef} className={menuClassName}>
+                    <ul>{stores}</ul>
+                </div>
+            </div>
+        ) : null;
+
+        if (hasMultipleStores) {
+            window.document.documentElement.style.setProperty(
+                '--header-height',
+                '7.5rem'
+            );
```

Comment:
There's gotta be a better way. Have you run the implementation by UX first and foremost? If this UX is what they want, then that means we need some way to conditionally use either `5rem` or `7.5rem`. Feels odd to do this in the component though.

Also, if we're going to keep this, we should use an effect, on mount, so that it does not continue to call the setter every render.